### PR TITLE
Adds the wizard's fireball spell

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Actions can now be aimable.
- Adds custom cursors. Holding shift swaps to a magnifying glass texture.
    - Aimable actions can specify a custom cursor.
- Adds the wizard's fireball spell. Continues #5160.
fireball
- Adds incendiary and explosive projectile components.
- Adds `SpriteHandler` and `DirectionalSpriteV2` components to some common mob prefabs.
- Bookshelves are now flammable.

CL:Adds the wizard's fireball spell. ONI'SOMA!

### CODE REVIEW

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Systems/Explosions/ExplosionComponent.cs  6
- UnityProject/Assets/Scripts/Systems/Spells/Wizard/Fireball.cs  1
- UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs  1
- UnityProject/Assets/Scripts/Items/Weapons/Projectiles/Behaviours/ProjectileIncendiary.cs  4
- UnityProject/Assets/Scripts/Items/Weapons/Projectiles/Behaviours/ProjectileExplode.cs  3
         

Complexity decreasing per file
==============================
+ UnityProject/Assets/Scripts/Systems/Spells/Wizard/TeleportAction.cs  -1
+ UnityProject/Assets/Scripts/NPC/AI/MobPathFinder.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6363081&bid=20652212)

File: UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs [hunk 1]
```diff
@@ -1,106 +1,220 @@
...
+			{
+				//Failfast
+				Logger.LogWarning($"Shoot request invalid, processing stopped: {ToString()}", Category.Firearms);
+				return;
+			}
+
+			//Not even spawned don't show bullets
+			if (PlayerManager.LocalPlayer == null) return;
+
+			LoadMultipleObjects(new uint[] { Shooter, Projectile });
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs:52]: Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6363081)

File: UnityProject/Assets/Scripts/Core/Input System/MouseInputController.cs [hunk 2]
```diff
@@ -587,4 +594,76 @@ private void ChangeDirection()
...
+
+		SetCursorTexture(texture, hotspot);
+	}
+
+	/// <summary>
+	/// Sets the cursor back to the system default.
+	/// </summary>
+	public static void ResetCursorTexture()
+	{
+		if (currentCursorTexture = null) return;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Input System/MouseInputController.cs:647]: Codacy found an issue: [Extract the assignment of 'currentCursorTexture' from this expression.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6363081)

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Core/Input System/MouseInputController.cs:647]: Nice catch.

File: UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs]: how do you feel about this one? I think we can ignore it

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs]: Oh, already force-pushed. Up to you.

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Systems/Spells/Wizard/Fireball.cs  1
- UnityProject/Assets/Scripts/Messages/Server/ShootMessage.cs  1
- UnityProject/Assets/Scripts/Systems/Explosions/ExplosionComponent.cs  6
- UnityProject/Assets/Scripts/Items/Weapons/Projectiles/Behaviours/ProjectileExplode.cs  3
- UnityProject/Assets/Scripts/Items/Weapons/Projectiles/Behaviours/ProjectileIncendiary.cs  4
         

Complexity decreasing per file
==============================
+ UnityProject/Assets/Scripts/NPC/AI/MobPathFinder.cs  -1
+ UnityProject/Assets/Scripts/Systems/Spells/Wizard/TeleportAction.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6363081&bid=20652212)

**Pull request merged**